### PR TITLE
hpmyroom: init at 11.1.0.0508

### PR DESCRIPTION
**Pull request by JohnAZoidberg**
###### Motivation for this change
To join a web conference instantiated by HPE's MyRoom this client can be used.
Everything seems to work, voice chat, screen sharing. It only doesn't show my webcam but I have never used that feature or know how (apart from the "Testing" screen).
Since I can't find a license, I classified it as nonfree and we can't build it with hydra and distribute it in the cache. 

There is a Darwin binary but I don't know how to patchelf there or whether it works there.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by worldofpeace** [comment]
@GrahamcOfBorg eval

**Comment by jpyen** [comment]
Tested on NixOS 19.03, works like a charm.

**Comment by JohnAZoidberg** [comment]
I think if you [create an account](https://www.myroom.hpe.com/Account/RegisterStart) you have access to a private room for testing.

**Comment by aanderse** [comment]
@worldofpeace seems like @JohnAZoidberg has done a good job on this (as always) as well as diligent testing... would you agree a merge seems to be in order?

**Comment by JohnAZoidberg** [comment]
Thanks :)
Rebased and updated to the latest version.

File: pkgs/applications/networking/hpmyroom/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  '';
+
+  meta = {
+    description = "Client for HPE's MyRoom web conferencing solution";
+    maintainers = with lib.maintainers; [ johnazoidberg ];
+    license = lib.licenses.unfree;
+    homepage = "https://myroom.hpe.com";
+    # TODO: A Darwin binary is available upstream
+    platforms = [ "x86_64-linux" ];
+    hydraPlatforms = [];
```
Comment (worldofpeace) [inline pkgs/applications/networking/hpmyroom/default.nix]: I don't think this is needed?

File: pkgs/applications/networking/hpmyroom/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  ];
+
+  buildInputs = let deps = [
+    libuuid libXtst libXScrnSaver libXfixes alsaLib freetype fontconfig libXext
+    libGL libpng libxcb libpulseaudio libdrm
+    glib  # For libgobject
+    stdenv.cc.cc  # For libstdc++
+  ] ++ (with gst_all_1; [ gstreamer gst-plugins-base ])
+    ++ (with xorg; [ libX11 ]);
+  in map (dep: dep.lib or dep.out or dep) deps;
```
Comment (worldofpeace) [inline pkgs/applications/networking/hpmyroom/default.nix]: Why is this needed? Seemed to work fine without it.

File: pkgs/applications/networking/hpmyroom/default.nix [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    rpmextract autoPatchelfHook makeWrapper
+  ];
+
+  buildInputs = let deps = [
+    libuuid libXtst libXScrnSaver libXfixes alsaLib freetype fontconfig libXext
+    libGL libpng libxcb libpulseaudio libdrm
+    glib  # For libgobject
+    stdenv.cc.cc  # For libstdc++
+  ] ++ (with gst_all_1; [ gstreamer gst-plugins-base ])
+    ++ (with xorg; [ libX11 ]);
```
Comment (worldofpeace) [inline pkgs/applications/networking/hpmyroom/default.nix]: This is unnecessary, please reference `xorg.libX11` directly.

**Review by worldofpeace — CHANGES_REQUESTED** [review]
The `substituteInPlace`, `wrapProgram`, and `ln` calls really belong in `postFixup`.
Is this a qt app needing a qt wrapper?

File: pkgs/applications/networking/hpmyroom/default.nix (hunk 1 above)
Comment (JohnAZoidberg) [reply pkgs/applications/networking/hpmyroom/default.nix]: I'm not sure. It cannot be redistributed so hydra must not build it. Is that given with `meta.license = lib.licenses.unfree;`?

**Comment by JohnAZoidberg** [comment]
Yes, it is a QT App. But it worked fine without (maybe I didn't try a feature that would have needed a library it cannot find without).

File: pkgs/applications/networking/hpmyroom/default.nix (hunk 1 above)
Comment (worldofpeace) [reply pkgs/applications/networking/hpmyroom/default.nix]: Yes, it will refuse to evaluate it when unfree. So it's unneeded.

File: pkgs/applications/networking/hpmyroom/default.nix [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  installPhase = ''
+    runHook preInstall
+
+    mv usr $out
+
+    runHook postInstall
+  '';
+
+  qtWrapperArgs = [
+    ''--prefix QT_XKB_CONFIG_ROOT : "${xorg.xkeyboardconfig}/share/X11/xkb"''
```
Comment (worldofpeace) [inline pkgs/applications/networking/hpmyroom/default.nix]:
```suggestion
    "--prefix QT_XKB_CONFIG_ROOT : '${xorg.xkeyboardconfig}/share/X11/xkb'"
```

**Comment by worldofpeace** [comment]
https://github.com/NixOS/nixpkgs/pull/63125/commits/f24ef244970bad6f0a72ac1855297343651a1c2c#r320353661 and I think we can merge.

**Comment by JohnAZoidberg** [comment]
Yup, that works, too. Thanks!
